google-cloud-compute/src/main/java/com/google/cloud/compute/testing/ResourceCleaner.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.compute.testing;
+
+import com.google.cloud.compute.AddressId;
+import com.google.cloud.compute.Compute;
+import com.google.cloud.compute.DiskId;
+import com.google.cloud.compute.ImageId;
+import com.google.cloud.compute.InstanceId;
+import com.google.cloud.compute.NetworkId;
+import com.google.cloud.compute.SnapshotId;
+import com.google.cloud.compute.SubnetworkId;
+import com.google.common.base.Function;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public class ResourceCleaner {
+  private final Compute compute;
+  private final Map<Object, ManagedResource<?>> resources;
+
+  private ResourceCleaner(Compute compute) {
+    this.compute = compute;
+    this.resources = new HashMap<>();
+  }
+
+  public static ResourceCleaner create(Compute compute) {
+    return new ResourceCleaner(compute);
+  }
+
+  private class ManagedResource<T> {
+    final T res;
+    Function<T, Void> func;
+
+    private ManagedResource(T res, Function<T, Void> func) {
+      this.res = res;
+      this.func = func;
+    }
+
+    public void delete() {
+      func.apply(res);
+    }
+  }
+
+  public void cleanUp() {
+    for (Map.Entry<Object, ManagedResource<?>> r : resources.entrySet()) {
+      System.out.println("Clean up compute engine resource: " + r.getKey().toString());
+      r.getValue().delete();
+    }
+    resources.clear();
+  }
+
+  public ResourceCleaner add(AddressId addressId) {
```

Comment:
I think we should remove the return value from the `add` methods since it's not used (and I don't see a use for it).